zaas-client/src/main/java/org/zowe/apiml/zaasclient/token/TokenServiceHttpsJwt.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.zaasclient.token;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HeaderElement;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.cookie.BasicClientCookie;
+import org.apache.http.util.EntityUtils;
+import org.zowe.apiml.zaasclient.client.HttpsClient;
+import org.zowe.apiml.zaasclient.exception.ZaasClientErrorCodes;
+import org.zowe.apiml.zaasclient.exception.ZaasClientException;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Slf4j
+public class TokenServiceHttpsJwt implements TokenService {
+    private static final String COOKIE_PREFIX = "apimlAuthenticationToken";
+    private final String loginEndpoint;
+    private final String queryEndpoint;
+    private final String host;
+
+    private HttpsClient httpsClient;
+
+    public TokenServiceHttpsJwt(HttpsClient client, String baseUrl, String host) {
+        this.httpsClient = client;
+        this.host = host;
+
+        loginEndpoint = baseUrl + "/login";
+        queryEndpoint = baseUrl + "/query";
+    }
+
```

Comment:
Are these endpoints stable? Can't be changed in future?

Reply:
There are APIML endpoints. In case any changes are made in APIML endpoints zaas-client should be modified to accommodate those changes.